### PR TITLE
[12.0][MIG] l10n_nl

### PR DESCRIPTION
**Pull request by astirpe**
Migration script for `l10n_nl` module.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_nl/migrations/12.0.3.0/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---Fields in module 'l10n_nl'---
+---XML records in module 'l10n_nl'---
+DEL account.account.tag: l10n_nl.tag_nl_13 (noupdate)
+DEL account.account.tag: l10n_nl.tag_nl_34 (noupdate)
```
Comment (pedrobaeza) [inline addons/l10n_nl/migrations/12.0.3.0/openupgrade_analysis_work.txt:4]: You should manually try to delete these records (with the openupgradelib method)

File: addons/l10n_nl/migrations/12.0.3.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        cr, """
+        INSERT INTO ir_model_data (module, name, model, res_id, noupdate)
+          SELECT 'l10n_nl',
+                 CONCAT('state_nl_', LOWER(s.code)),
+                 'res.country.state',
+                 s.id,
+                 false
+          FROM res_country_state s
+              JOIN res_country c ON s.country_id = c.id
+              WHERE c.code LIKE 'NL'
```
Comment (pedrobaeza) [inline addons/l10n_nl/migrations/12.0.3.0/pre-migration.py:17]: I think you should limit possible states here (as there can be manual entries created even without codes). Something like:
```
AND s.code IN ('bq1', 'bq2', 'bq3', 'dr'...)
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by MiquelRForgeFlow** [comment]
Why don't load a `noupdate_changes.xml` file?

with:

```xml
<?xml version='1.0' encoding='utf-8'?>
<odoo>
  <record id="tag_nl_40" model="account.account.tag">
    <field name="name">3b. Leveringen naar of diensten in landen binnen de EU (omzet) (producten)</field>
  </record>
  <record id="tag_nl_10" model="account.account.tag">
    <field name="name">2a. Leveringen/diensten waarbij de omzetbelasting naar u is verlegd (omzet)</field>
  </record>
  <record id="tag_nl_27" model="account.account.tag">
    <field name="name">2a. Leveringen/diensten waarbij de omzetbelasting naar u is verlegd (BTW)</field>
  </record>
  <record id="tag_nl_23" model="account.account.tag">
    <field name="name">1c. Leveringen/diensten belast met overige tarieven, behalve 0% (BTW)</field>
  </record>
  <record id="tag_nl_41" model="account.account.tag">
    <field name="name">3b. Leveringen naar of diensten in landen binnen de EU (omzet) (diensten)</field>
  </record>
  <record id="tag_nl_06" model="account.account.tag">
    <field name="name">1c. Leveringen/diensten belast met overige tarieven, behalve 0% (omzet)</field>
  </record>
</odoo>
```

**Comment by astirpe** [comment]
@mreficent the changed tag names were introduced with https://github.com/odoo/odoo/commit/146e0692e538110caf1a2be5ff5ce38da17a311f to fix https://github.com/odoo/odoo/pull/30539 in order to match the legal Dutch tag names. So it seems to me that they did a bug fixing. I would keep those tags renamed the V12 way, since I don't see any potential issue in that. What do you think?

**Comment by MiquelRForgeFlow** [comment]
What you say it's ok to me. I don't have a strong opinion here.

File: addons/l10n_nl/migrations/12.0.3.0/openupgrade_analysis_work.txt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+DEL account.account.template: l10n_nl.transfer_account_id
+# DONE: post-migration: deleted safely
```
Comment (pedrobaeza) [inline addons/l10n_nl/migrations/12.0.3.0/openupgrade_analysis_work.txt:5]: There's no need of deleting this not being noupdate=1

File: addons/l10n_nl/migrations/12.0.3.0/openupgrade_analysis_work.txt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+NEW account.fiscal.position.tax.template: l10n_nl.position_tax_eu_no_taxes_report_7
+NEW account.fiscal.position.tax.template: l10n_nl.position_tax_eu_no_taxes_report_8
+NEW account.fiscal.position.tax.template: l10n_nl.position_tax_eu_no_taxes_report_9
+NEW account.fiscal.position.template: l10n_nl.fiscal_position_template_eu_no_taxes_report
+NEW account.tax.template: l10n_nl.btw_X0_diensten
+NEW account.tax.template: l10n_nl.btw_X0_producten
+# NOTHING TO DO
+
+DEL account.tax.template: l10n_nl.btw_X0
+# DONE: post-migration: deleted safely
```
Comment (pedrobaeza) [inline addons/l10n_nl/migrations/12.0.3.0/openupgrade_analysis_work.txt]: There's no need of deleting this not being noupdate=1

**Review by pedrobaeza — APPROVED** [review]

**Comment by astirpe** [comment]
@mreficent can this one be merged for you?

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Pull request merged**